### ISSUE TITLE
simple_kube demo doesn't work

### ISSUE THREAD

**Issue by krancour** [hashicorp/da-connect-demo#5]
consul: v1.3.0
consul-k8s: v0.2.0

These were installed using the master branch of the helm chart. The only tweak I made to values was to enable sidecar injection.

Following the directions for the simple_kube to the letter, the thing I am observing is that the following does not seem to be honored:

```
"consul.hashicorp.com/connect-service-upstreams": "demo-website:8000,demo-api:8001"
```

The result is that these upstreams in the Nginx config are invalid:

```
upstream demo-website {
upstream demo-website {
  server localhost:8000;
}
upstream demo-api {
  server localhost:8001;
}
```

Requests to Nginx therefore result in 502 bad gateway.

If I exec into the pod, `netstat` shows me nothing is bound to 127.0.0.1:8000 and 127.0.0.1:8001 as I would expect:

```
$ k exec -it demo-ingress-8578f5c8c7-2jts8 -c consul-connect-envoy-sidecar -- netstat -tulpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      -
tcp        0      0 127.0.0.1:19000         0.0.0.0:*               LISTEN      1/envoy
```

Envoy _is_ listening on 127.0.0.1, although I'm not sure what for.

**Comment by krancour** [comment]
New datapoint: this failure only seems to occur when Consul is installed in a namespace other than `default`. You would need to install it elsewhere to re-create this issue, but I am able to recreate it reliably on both AKS and minikube.

fwiw, I don't see any obvious issues in init container logs or envoy logs.

**Comment by krancour** [comment]
And now I'm back to having this not work regardless of namespace. I'm not sure what was different the one time I lucked into this working.

**Comment by krancour** [comment]
Ok... got to the bottom of this.

I had installed / deleted / re-installed the Consul Helm chart a number of times in the `consul-system` namespace. Things weren't working. (Specifically, I discovered that service registrations were failing silently.)

In a last-ditch attempt at troubleshooting, I deleted and re-installed into the `default` namespace-- wondering if registration had somehow been impaired by some poor assumption or bit of hardcoding vis a vis namespace. That was when I saw things start working... but subsequent uninstalls / re-installs of Consul to the `default` namespace resulted in things _not_ working again.

I started to wonder if the factor for success was simply being in a _new_ namespace. It turned out it was. So it must have been that some state was unexpectedly being left behind... it was...

What I failed to realize was that when `helm delete consul --purge` deleted the `StatefulSet` for the Consul server, it _does not_ cascade that delete down to PVCs...

So, in any case where I deleted and re-installed in a namespace that already contained those PVCs, I was effectively launching a _new_ cluster of Consul servers backed by _old_ data. I'm not clear on why that doesn't work (one would think it should), but it doesn't.

If I take care to manually delete PVCs (or the entire namespace) a `helm delete consul --purge`, it gets me back on track for the next time I re-install into the same namespace.

So the examples are clearly not the issue here. I'll be closing this issue.

But I will open an issue against the consul-helm project because, imo, re-installing with existing PVCs _should_ work and, if for some reason, it really ought not to, then this behavior should be documented.
